BookReader/plugins/plugin.tts.js.map

```diff
@@ -0,0 +1 @@
 number, number]} DJVURect\n * coords are in l,b,r,t order\n */\n\n/**\n * @template TSound extra properties added to PageChunk to allow playing\n */\nexport default class AbstractTTSEngine {\n    /**\n     * @protected\n     * @param {TTSEngineOptions} options \n     */\n    constructor(options) {\n        this.playing = false;\n        this.opts = options;\n        /** @type {AsyncStream<PageChunk>} */\n        this.chunkStream = null;\n        /** Different engines should overwrite this */\n        this.isSupported = false;\n    }\n\n    /** @abstract */\n    init() { return null; }\n\n    /**\n     * @param {number} leafIndex\n     * @param {number} numLeafs total number of leafs in the current book\n     */\n    start(leafIndex, numLeafs) {\n        this.playing = true;\n        this.opts.onLoadingStart();\n\n        /** @type {AsyncStream<PageChunk>} */\n        this.chunkStream = AsyncStream.range(leafIndex, numLeafs-1)\n        .map(this.fetchPageChunks.bind(this))\n        .buffer(2)\n        .flatten();\n\n        this.step();\n    }\n\n    stop() {\n        this.playing = false;\n        this.chunkStream = null;\n    }\n\n    /**\n     * @private\n     */\n    step() {\n        this.getPlayStream().pull()\n        .then(item => {\n            if (item.done) {\n                this.stop();\n                this.opts.onDone();\n                return;\n            }\n            this.opts.onLoadingComplete();\n            return this.opts.beforeChunkPlay(item.value).then(() => item)\n        })\n        .then(item => {\n            if (this.playing) {\n                return this.playChunk(item.value);\n            }\n        })\n        .then(() => {\n            if (this.playing) return this.step();\n        });\n    }\n\n    /**\n\n     * @abstract\n     * @return {AbstractStream<PageChunk & TSound>}\n     */\n    getPlayStream() { return null; }\n\n    /**\n     * @abstract\n     * @param {PageChunk & TSound} chunk\n     * @return {PromiseLike} promise called once playing finished\n     */\n    playChunk(chunk) { return null; }\n\n    /**\n     * @private\n     * Gets the text on a page with given index\n     * @param {Number} leafIndex\n     * @return {PromiseLike<Array<PageChunk>>}\n     */\n    fetchPageChunks(leafIndex) {\n        return $.ajax({\n            type: 'GET',\n            url: 'https://'+this.opts.server+'/BookReader/BookReaderGetTextWrapper.php',\n            dataType:'jsonp',\n            data: {\n                path: this.opts.bookPath+'_djvu.xml',\n                page: leafIndex\n            }\n        })\n        .then(\n            /** @param {Array<[String, ...Array<DJVURect>]>} chunks */\n            function (chunks) {\n                return chunks.map(c => {\n                    return {\n                        leafIndex,\n                        text: c[0],\n                        lineRects: c.slice(1)\n                    };\n                });\n            }\n        );\n    }\n}"],"sourceRoot":""}
```

Comment:
Can we generate the map file at build/deploy time instead of checking it into the repo?

Reply:
This is one of the bigger questions I have about the approach I'm using. I'm going to post a response to the full PR for better visibility.

Reply:
As discussed; map file is now vanquished \o/